frontend/src/component/SignUpForm.js

```diff
@@ -0,0 +1,137 @@
+import React, { useEffect } from "react";
+import { useSelector } from "react-redux";
+import { useHistory } from "react-router";
+import { Field, Form } from "react-final-form";
+import api from "../api/api";
+
+const SignUpForm = () => {
+  const auth = useSelector((state) => state.auth);
+
+  const history = useHistory();
+  const required = (value) => (value ? undefined : "this field is required");
+  const validatePasswordMatch = (values) => {
+    const errors = {};
+    if (values.password !== values.password2) {
+      errors.password2 = "Password does not match";
+    }
+    return errors;
+  };
+
+  useEffect(() => {
+    if (auth.isLoggedIn) {
+      history.push("/");
+    }
+  }, [auth.isLoggedIn, history]);
+
+  const signUp = (formValues) => {
+    api
+      .post("/signup/", formValues)
+      .then((res) => {
+        history.push("/signin");
```

Comment:
First method call should be inline with object
```suggestion
    api.post("/signup/", formValues)
      .then((res) => {
        history.push("/signin");
```